packages/typegen/src/write/sql.ts

```diff
@@ -122,7 +122,7 @@ export function getSQLHelperContent(query: TaggedQuery, destPath: string) {
       if (cached) {
         return cached
       }
-      const content = fs.readFileSync(filepath).toString()
+      const content = fs.readFileSync(filepath).toString().replace(/\\$(\\d)/g, '#$#$1')
```

Comment:
@gajus it'd be great to get your input on what to do here - basically some more context on what I'm talking about in https://github.com/gajus/slonik/issues/320#issuecomment-997380918. What's happening:

- User writes `select * from foo where bar = $1` in a file called `foobar.sql`.
- User runs slonik-typegen and a `foobar.sql.ts` file is generated containing this code
-  The generated module can then be imported with something like:

```ts
import * as fooBarQueries from './foobar.sql'

export default async () => {
   return pool.query(fooBarQueries.getFooBarQuerySync())
}
```

The problem is, under the hood `getFooBarQuerySync` now needs to add this regex replace to convert `$1` into `#$#1`. There are two problems with that:

1. It's brittle. It will do the wrong thing for queries like `select * from foo where bar = $1 or baz = '$1'`
2. It will _only_ work for slonik >=26.1.0